pkg/images/enricher/metrics.go

```diff
@@ -72,13 +95,27 @@ func newMetrics(subsystem pkgMetrics.Subsystem) metrics {
 			Help:      "Amount of time it's taken to retrieve vulns for an image in ms",
 			Buckets:   prometheus.ExponentialBuckets(4, 2, 16),
 		}, []string{"Scanner", "Error"}),
+		enricherSemaphoreQueueSize: prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: pkgMetrics.PrometheusNamespace,
+			Subsystem: subsystem.String(),
+			Name:      "enricher_semaphore_queue_size",
+			Help:      "A counter that tracks the size of the queues for the scan semaphores used in image scans.",
```

Comment:
This is not a counter!

Reply:
Those two extra metrics could also be another label for the previous metric. Something like: `location=enricher`, whereas in the other case it would be `location=image-scan`. That would make it more compact and easier to find in the list of metrics.

That comment with labels make sense only for the metrics that are in the same origin: i.e., one metric with labels for Sensor, another metric with labels for Central.

Reply:
> Those two extra metrics could also be another label for the previous metric. Something like: location=enricher, whereas in the other case it would be location=image-scan

I like this idea but I'm having some trouble implementing this. The 3 sets of metrics I'm adding are in very different parts of the code, one in central, one in sensor, one in a package. Is it really possible to combine them all into a few `GaugeVec`s or am I misunderstanding what you're saying I should try?

Reply:
You can define the metrics in different places as long as they have the same name. Just make sure to separate them by label.

Reply:
I think, this has been addressed by Jan. Resolving.